lib/game_logic.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+class GameLogic
+  attr_reader :player1_array, :player2_array, :wining_compositions
+  attr_accessor :winner, :cells, :player1, :player2, :current_player, :mark
+  def initialize
+    @wining_compositions = [[1, 2, 3], [4, 5, 6], [7, 8, 9], [1, 4, 7], [2, 5, 8], [3, 6, 9], [1, 5, 9], [3, 5, 7]]
+    @winner = nil
+    @player1 = nil
+    @player2 = nil
+    @current_player = nil
+    @player1_array = []
+    @player2_array = []
+    @cells = [1, 2, 3, 4, 5, 6, 7, 8, 9]
+  end
+
+  def check_winner
+    for i in 0...8
```

Comment:
You can use `each` to fix linter warning/error.

```suggestion
    (0...8).each do |i|
```